src/Native/System.Native/pal_networking.cpp

```diff
@@ -0,0 +1,332 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_config.h"
+#include "pal_networking.h"
+#include "pal_utilities.h"
+
+#include <arpa/inet.h>
+#include <assert.h>
+#include <functional>
+#include <netdb.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+#include <unistd.h>
+#include <vector>
+
+const int NUM_BYTES_IN_IPV4_ADDRESS = 4;
+const int NUM_BYTES_IN_IPV6_ADDRESS = 16;
+
+static_assert(PAL_HOST_NOT_FOUND == HOST_NOT_FOUND, "");
+static_assert(PAL_TRY_AGAIN == TRY_AGAIN, "");
+static_assert(PAL_NO_RECOVERY == NO_RECOVERY, "");
+static_assert(PAL_NO_DATA == NO_DATA, "");
+static_assert(PAL_NO_ADDRESS == NO_ADDRESS, "");
+
+static void IpStringToAddressHelper(const char* address, const char* port, bool isIpV6, int32_t* err, std::function<void(const addrinfo& info)> lambda)
+{
+    assert(address != nullptr);
+    assert(err != nullptr);
+
+    addrinfo hint = {};
+    hint.ai_flags = AI_NUMERICHOST | AI_NUMERICSERV;
+    hint.ai_family = isIpV6 ? AF_INET6 : AF_INET;
+    hint.ai_socktype = 0;
+    hint.ai_protocol = 0;
+
+    addrinfo* info = nullptr;
+    int result = getaddrinfo(address, port, &hint, &info);
+    if (result == 0)
+    {
+        *err = 0;
+        lambda(*info);
```

Comment:
No, because the IPv6 version needs to pass in a port where as the IPv4 version needs to pass a port out; also, in IPv6, the scope needs to be passed out whereas scope doesn't exist in IPv4. Having one function to do all of this looked messy and makes the function multi-use and less clear of what parameter is used where and by whom.